synapse/rest/media/v1/_base.py

```diff
@@ -287,44 +289,25 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         pass
 
 
+@attr.s(slots=True, auto_attribs=True)
 class FileInfo:
-    """Details about a requested/uploaded file.
-
-    Attributes:
-        server_name (str): The server name where the media originated from,
-            or None if local.
-        file_id (str): The local ID of the file. For local files this is the
-            same as the media_id
-        url_cache (bool): If the file is for the url preview cache
-        thumbnail (bool): Whether the file is a thumbnail or not.
-        thumbnail_width (int)
-        thumbnail_height (int)
-        thumbnail_method (str)
-        thumbnail_type (str): Content type of thumbnail, e.g. image/png
-        thumbnail_length (int): The size of the media file, in bytes.
-    """
-
-    def __init__(
-        self,
-        server_name,
-        file_id,
-        url_cache=False,
-        thumbnail=False,
-        thumbnail_width=None,
-        thumbnail_height=None,
-        thumbnail_method=None,
-        thumbnail_type=None,
-        thumbnail_length=None,
-    ):
-        self.server_name = server_name
-        self.file_id = file_id
-        self.url_cache = url_cache
-        self.thumbnail = thumbnail
-        self.thumbnail_width = thumbnail_width
-        self.thumbnail_height = thumbnail_height
-        self.thumbnail_method = thumbnail_method
-        self.thumbnail_type = thumbnail_type
-        self.thumbnail_length = thumbnail_length
+    """Details about a requested/uploaded file."""
+
+    # The server name where the media originated from, or None if local.
+    server_name: Optional[str]
+    # The local ID of the file. For local files this is the same as the media_id
+    file_id: str
+    # If the file is for the url preview cache
+    url_cache: Optional[str] = None
```

Comment:
As far as I can tell `url_cache` is supposed to be a string. It is saved in the tables as `TEXT`, but we were previously sometimes setting it to a boolean. 🤷

Reply:
I'm not seeing anywhere it is used as a string - am I blind? Indeed, as far as I can tell, the only place it is read is in `MediaStorage._file_info_to_path`, which just uses its boolean value.

There's a `url_cache` in `local_media_repository`, but it doesn't seem to be populated from a `FileInfo`.

Reply:
We pull info from `local_media_repository` and then create a `FileInfo` in one spot:

https://github.com/matrix-org/synapse/blob/63f28e4a0c22c8133a6da175c7986e389fc781e2/synapse/rest/media/v1/media_repository.py#L213

Maybe that should be turning the `str` into a `bool` and the rest should be left alone?

Reply:
It also seems to happen when responding with local thumbnails:

https://github.com/matrix-org/synapse/blob/63f28e4a0c22c8133a6da175c7986e389fc781e2/synapse/rest/media/v1/thumbnail_resource.py#L117

Which calls into:

https://github.com/matrix-org/synapse/blob/63f28e4a0c22c8133a6da175c7986e389fc781e2/synapse/rest/media/v1/thumbnail_resource.py#L346

Reply:
right yes, I think we should be casting to a bool in those places.